official/vision/beta/projects/yolo/modeling/backbones/darknet.py

```diff
@@ -319,86 +505,135 @@ def _csp_stack(self, inputs, config, name):
-                                   **self._default_dict)(inputs)
-    self._default_dict["activation"] = self._activation
-    self._default_dict["name"] = None
+    self._default_dict['activation'] = self._get_activation(config.activation)
+    self._default_dict['name'] = f'{name}_csp_tiny'
+    x, x_route = nn_blocks.CSPTiny(
+        filters=config.filters, **self._default_dict)(
+            inputs)
+    self._default_dict['activation'] = self._activation
+    self._default_dict['name'] = None
     return x, x_route
 
   def _tiny_stack(self, inputs, config, name):
-    x = tf.keras.layers.MaxPool2D(pool_size=2,
-                                  strides=config.strides,
-                                  padding="same",
-                                  data_format=None,
-                                  name=f"{name}_tiny/pool")(inputs)
-    self._default_dict["activation"] = self._get_activation(config.activation)
-    self._default_dict["name"] = f"{name}_tiny/conv"
+    x = tf.keras.layers.MaxPool2D(
+        pool_size=2,
+        strides=config.strides,
+        padding='same',
+        data_format=None,
+        name=f'{name}_tiny/pool')(
+            inputs)
+    self._default_dict['activation'] = self._get_activation(config.activation)
+    self._default_dict['name'] = f'{name}_tiny/conv'
     x = nn_blocks.ConvBN(
         filters=config.filters,
         kernel_size=(3, 3),
         strides=(1, 1),
-        padding="same",
+        padding='same',
         **self._default_dict)(
             x)
-    self._default_dict["activation"] = self._activation
-    self._default_dict["name"] = None
+    self._default_dict['activation'] = self._activation
+    self._default_dict['name'] = None
     return x
 
   def _residual_stack(self, inputs, config, name):
-    self._default_dict["activation"] = self._get_activation(config.activation)
-    self._default_dict["name"] = f"{name}_residual_down"
-    x = nn_blocks.DarkResidual(filters=config.filters,
-                               downsample=True,
-                               **self._default_dict)(inputs)
-    for i in range(config.repetitions - 1):
-      self._default_dict["name"] = f"{name}_{i}"
-      x = nn_blocks.DarkResidual(filters=config.filters,
-                                 **self._default_dict)(x)
-    self._default_dict["activation"] = self._activation
-    self._default_dict["name"] = None
+    self._default_dict['activation'] = self._get_activation(config.activation)
+    self._default_dict['name'] = f'{name}_residual_down'
+    if self._dilate:
+      self._default_dict['dilation_rate'] = config.dilation_rate
+      if config.repetitions < 8:
+        config.repetitions += 2
+    else:
+      self._default_dict['dilation_rate'] = 1
+
+    x = nn_blocks.DarkResidual(
+        filters=config.filters, downsample=True, **self._default_dict)(
+            inputs)
+
+    dilated_reps = config.repetitions - self._default_dict[
+        'dilation_rate'] // 2 - 1
```

Comment:
I would use parentheses to clarify the order of operations

Reply:
Fixed